cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -506,98 +429,129 @@ func (s *Server) setupRepoKubeappsManagedSecrets(
+	// for now: we disallow updates to pending repos and allow them for non-pending ones
+	// (i.e. success or failed status)
+	complete, _, _ := isHelmRepositoryReady(*repo)
+	if !complete {
+		return nil, status.Errorf(codes.Internal, "updates to repositories pending reconciliation are not supported")
 	}
 
-	if certFile, ok := secret.Data["certFile"]; ok {
-		if keyFile, ok := secret.Data["keyFile"]; ok {
-			auth.Type = corev1.PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_TLS
-			auth.PackageRepoAuthOneOf = &corev1.PackageRepositoryAuth_TlsCertKey{
-				TlsCertKey: &corev1.TlsCertKey{
-					Cert: string(certFile),
-					Key:  string(keyFile),
-				},
-			}
-		}
-	} else if username, ok := secret.Data["username"]; ok {
-		if pwd, ok := secret.Data["password"]; ok {
-			auth.Type = corev1.PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_BASIC_AUTH
-			auth.PackageRepoAuthOneOf = &corev1.PackageRepositoryAuth_UsernamePassword{
-				UsernamePassword: &corev1.UsernamePassword{
-					Username: string(username),
-					Password: string(pwd),
-				},
-			}
-		}
-	} else if configStr, ok := secret.Data[".dockerconfigjson"]; ok {
-		dc, err := common.SecretDataToDockerCredentials(string(configStr))
-		if err != nil {
-			return nil, nil, err
+	if url == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "repository url may not be empty")
+	}
+	repo.Spec.URL = url
+
+	// flux does not grok description yet
+
+	if interval > 0 {
+		repo.Spec.Interval = metav1.Duration{Duration: time.Duration(interval) * time.Second}
+	} else {
+		// interval is a required field
+		repo.Spec.Interval = defaultPollInterval
+	}
+
+	if tlsConfig != nil && tlsConfig.InsecureSkipVerify {
+		return nil, status.Errorf(codes.InvalidArgument, "TLS flag insecureSkipVerify is not supported")
+	}
+
+	var secretRef string
+	if s.pluginConfig.UserManagedSecrets {
+		if secretRef, err = s.validateRepoUserManagedSecrets(ctx, key, tlsConfig, auth); err != nil {
+			return nil, err
 		}
-		auth.Type = corev1.PackageRepositoryAuth_PACKAGE_REPOSITORY_AUTH_TYPE_DOCKER_CONFIG_JSON
-		auth.PackageRepoAuthOneOf = &corev1.PackageRepositoryAuth_DockerCreds{
-			DockerCreds: dc,
+	} else {
+		if secretRef, err = s.updateRepoKubeappsManagedSecrets(ctx, key, tlsConfig, auth, repo.Spec.SecretRef); err != nil {
+			return nil, err
 		}
+	}
+
+	if secretRef != "" {
+		repo.Spec.SecretRef = &fluxmeta.LocalObjectReference{Name: secretRef}
 	} else {
-		log.Warning("Unrecognized type of secret [%s]", secret.Name)
+		repo.Spec.SecretRef = nil
 	}
-	return tlsConfig, auth, nil
+
+	repo.Spec.PassCredentials = auth != nil && auth.PassCredentials
+
+	// get rid of the status field, since now there will be a new reconciliation
+	// process and the current status no longer applies. metadata and spec I want
+	// to keep, as they may have had added labels and/or annotations and/or
+	// even other changes made by the user.
+	repo.Status = sourcev1.HelmRepositoryStatus{}
```

Comment:
Oh interesting... is that necessary? Won't it get overwritten when the new reconciliation process begins anyway? Interested to know if this is what the kubectl client does when updating something, or why it's otherwise needed here?

Reply:
> Oh interesting... is that necessary? Won't it get overwritten when the new reconciliation process begins anyway? Interested to know if this is what the kubectl client does when updating something, or why it's otherwise needed here?

eeeh, its been too long since I wrote this code (similar done for helmreleases) that I am struggling to remember the exact reason I put that in. I do remember same reasoning that you mentioned but something (I don't remember what) made me add this line in. It may have had something to do with unit tests (i.e. no real flux reconciler involved). Don't know off the top of my head if there is a solid reason for it to be in production. I know that its one line and its not hurting anything. If you insist, I will remove it and see what's what.